sdks/java/core/src/main/java/org/apache/beam/sdk/io/LocalFileSystem.java

```diff
@@ -225,7 +225,11 @@ private MatchResult matchOne(String spec) throws IOException {
     }
 
     File parent = file.getAbsoluteFile().getParentFile();
-    if (!parent.exists()) {
+    while (parent != null && !parent.exists()) {
```

Comment:
If the path is relative then can you stop search at the current directory instead of continuing all the way to the root?

If the path is absolute, can you use something like this [non glob part matcher found in GcsUtil](https://github.com/apache/beam/blob/de18f4b34c8eed1e7ce0d62a17e99c47b8c2f8b4/sdks/java/extensions/google-cloud-platform-core/src/main/java/org/apache/beam/sdk/extensions/gcp/util/GcsUtil.java#L156)?

Reply:
If you are OK with me using a regex for glob I can make this whole thing simpler, will change